### PR TITLE
[Issue-1509] Change permission request for Photo and Camera

### PR DESCRIPTION
**Pull request by BjornVanslembrouck**
### Description of Change ###
Explicitly check and ask for permissions as since iOS 11 the UIImagePickerController normally takes care of this. This however can lead to the camera view being open while a user denies the permission and it just showing a black screen. Might be less confusing if it lets the user know that the permission was not granted so they can ask for it again. 

Divided the checks in to camera and photo separately. It could be somebody blocking access to their Photo Library but wanting to access the camera. Therefor check what action they want to perform and check the permission accordingly.

### Bugs Fixed ###

- Related to issue #1509 

### API Changes ###

None

### Behavioral Changes ###

Instead of seeing the UIImagePickerController appearing behind the alert asking for permission and potentially resulting in a black screen, it now asks for the permission before opening UIImagePickerController

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [ ] Has samples (if omitted, state reason in description)
- [X] Rebased on top of `main` at time of PR
- [X] Changes adhere to coding standard
- [ ] Updated documentation ([see walkthrough](https://github.com/xamarin/Essentials/wiki/Documenting-your-code-with-mdoc))

### CODE REVIEW

File: Xamarin.Essentials/MediaPicker/MediaPicker.ios.cs [hunk 1]
```diff
@@ -40,11 +40,11 @@ static async Task<FileResult> PhotoAsync(MediaPickerOptions options, bool photo,
             if (!photo)
                 await Permissions.EnsureGrantedAsync<Permissions.Microphone>();
 
-            // permission is not required on iOS 11 for the picker
-            if (!Platform.HasOSVersion(11, 0))
```
Comment (mattleibow) [inline Xamarin.Essentials/MediaPicker/MediaPicker.ios.cs:44]: Is there a reason why this line was removed? It shows a popup on newer version which basically ask whether the app can access the specific photo or all the photos. We are never going to need all the photos, so we don't need to ask in this case.

**Comment by mattleibow** [comment]
Thanks for this PR. This change is very good, so thanks for taking the time.

File: Xamarin.Essentials/MediaPicker/MediaPicker.ios.cs (hunk 1 above)
Comment (BjornVanslembrouck) [reply Xamarin.Essentials/MediaPicker/MediaPicker.ios.cs:44]: @mattleibow, to be honest, I didn't really understood why it was there in the first place. Looks like you only wanted to verify the permissions on devices lower than iOS11 and that only for photo's and not the camera. 
I assumed permissions should be asked on all supported iOS versions (I tested from 10.3 to 14.1) and it seem to be doing the same on all of them.

I know that when opening a `UIImagePickerController` on later iOS versions, it asks for the permissions when opening the controller, but that could potentially lead to the same problems as opening the camera, denying the permissions and only seeing a black screen that has to be dismissed manually.

The way I implemented this, you are always sure that the permissions you need are granted before even trying to pick or capture media.

If I'm missing something, please do let me know.

Comment (mattleibow) [reply Xamarin.Essentials/MediaPicker/MediaPicker.ios.cs:44]: Ah, with the photo picker, the permissions are automatically granted on the newer devices so we don't have to explicitly ask for them.

Basically, for the picker, we need to ask before 11, but for the camera, we need to always ask.

Comment (BjornVanslembrouck) [reply Xamarin.Essentials/MediaPicker/MediaPicker.ios.cs:44]: I see, I will admit I had to read the documentation a couple of times to understand it properly...
Updated the PR to ask for photo permissions when it is `pickExisting AND < iOS11`, but always ask if it is `!pikcExisting`

**Pull request merged**